internal/state/store.go

```diff
@@ -122,8 +122,12 @@ type dbStore struct {
 var _ Store = (*dbStore)(nil)
 
 // NewStore creates the dbStore of the state pkg.
-func NewStore(db dbm.DB) Store {
-	return dbStore{db, log.NewNopLogger()}
+func NewStore(db dbm.DB, logger ...log.Logger) Store {
```

Comment:
Please explain with a comment in the code what is happening here

Reply:
done